File: backend-csharp/DTOs/Requests/RevokeTokenRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Picture2Text.Api.DTOs.Requests;

/// <summary>
/// 撤銷 Token 請求
/// </summary>
public class RevokeTokenRequest
{
    /// <summary>
    /// 要撤銷的 Refresh Token（可選，如果不提供則撤銷當前使用者的所有 token）
    /// </summary>
    public string? RefreshToken { get; set; }

    /// <summary>
    /// 是否撤銷所有裝置的 token
    /// </summary>
    public bool RevokeAllDevices { get; set; } = false;
}

File: backend-nestjs/src/dto/request/revoke-token.request.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { IsBoolean, IsOptional, IsString } from 'class-validator';

export class RevokeTokenRequest {
  @ApiPropertyOptional({
    description: 'Refresh Token（如果不提供且已登入，將從 session 撤銷）',
    example: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...',
  })
  @IsOptional()
  @IsString({ message: 'Refresh Token 必須是字串' })
  RefreshToken?: string;

  @ApiPropertyOptional({
    description: '是否撤銷所有裝置的 Token',
    example: false,
    default: false,
  })
  @IsOptional()
  @IsBoolean({ message: 'RevokeAllDevices 必須是布林值' })
  RevokeAllDevices?: boolean;
}
